Update procon for venice unleashed

FlashHit opened the pull request:
I don't know if I should split this in separate pull requests or not. I will just put all in here and if you want it splitted it will split it up for you.

So the first thing is that ea guids don't start with EA_ in VU.  so banning in the banlist tab is not working. Ea guids in VU have 32digits.

The next thing are the default VU rcon commands: 
https://docs.veniceunleashed.net/hosting/commands/
https://github.com/EmulatorNexus/VU-Docs/issues/12

I think each of them should be in the PRoConPluginAPI so we have an event when one of it gets triggered to create new plugins.

And then at least there is a mod that brings back textChatModeration.* and makes use of the undocumented unlockList.*. It also provides all VU server events for rcon. I.e. player.onRevive player.onKitPickup and a bunch of other stuff. https://github.com/FlashHit/AdvancedRCON 
This is a bit more custom but adding all these events would make it possible to create many other plugins with procon.
Unlocking the textChatModeration tab and spectator tab for VU would be also cool. 

I already made a custom version of procon for myself where I implemented all of it while it it is still compatible with all other bf games but it's already outdated since there were a couple of updates for procon and the modList.* commands were changed a bit. https://github.com/FlashHit/PRoCon-VU-Edition
Maybe I fork the new version later and update it.

Commit:
Upgrade to .NET 4.7, C# 7.1, roslyn

- Upgrade of Visual Studio solution file
- Upgrade of target framework to .NET Framework 4.7
- Upgrade of language version to latest minor version (C# 7.1)
- Changed plugin compiler from csc to roslyn
- Removed warnings and deprecations

Commit:
Reverted AddHostEvidence

- Reverted this change, due to compatibility with mono

Commit:
Fixed roundban information

Commit:
Added support for partial plugin classes

Prophet731 committed:
Update README.md

Updated the links

Prophet731 committed:
Updated URL for update check

Prophet731 committed:
Updated link

Prophet731 committed:
Updated links in about section

Prophet731 committed:
Updated URL

Prophet731 committed:
Incremented plugin version

Prophet731 committed:
Updates

Prophet731 committed:
Merge pull request #1 from AdKats/develop

Version 1.5.2.1

Prophet731 committed:
Wrapped in try..catch

Prophet731 committed:
Updated name

Prophet731 committed:
Updated link

Prophet731 committed:
Create dotnetcore.yml

Prophet731 committed:
Develop (#2)

* Wrapped in try..catch

* Updated name

* Updated link

Prophet731 committed:
Merge branch 'master' of https://github.com/AdKats/Procon-1

Prophet731 committed:
Updated .gitignore

Prophet731 committed:
Updated

Prophet731 committed:
Does this work for CI testing now?

Prophet731 committed:
Updated README and removed github workflows

ColColonCleaner committed:
Merge branch 'master' into dotnet-4.7

ColColonCleaner committed:
Merge pull request #3 from AdKats/dotnet-4.7

Dotnet 4.7

Prophet731 committed:
Updated RSS feed url

Prophet731 committed:
Updated URL

Prophet731 committed:
Updated URL

Prophet731 committed:
Created global suppressions file

Prophet731 committed:
Created global suppressions file

Prophet731 committed:
Updated obsoliete function and replaced with new one.

Prophet731 committed:
Suppressed warnings of an obsolete function from the old days

Prophet731 committed:
Revert "Updated obsoliete function and replaced with new one."

This reverts commit 19c651e2fb7695fa53c157ad46c007340f1e9928.

Prophet731 committed:
Updated links to the myrcon pdf

Prophet731 committed:
Code cleanup

Prophet731 committed:
Updated version

Prophet731 committed:
Updated BFHL def

Prophet731 committed:
Updated the RSS feed url and incremented version to 1.5.3.1

susch19 committed:
Fixed double dll pluginClassName (#4)

CSharpCompilation shouldn't include dll for assemblyname, because otherwise the referenced assembly is names *.dll.dll

Prophet731 committed:
Updated version number

Prophet731 committed:
Typo

Prophet731 committed:
Procon Version 1.5.3.3

Prophet731 committed:
Didn't update readme

Prophet731 committed:
More updates

Prophet731 committed:
Version 1.5.3.4

Prophet731 committed:
Failed to build everything

Mik4sa commented:
I think you should open pull requests to https://github.com/AdKats/Procon-1 now. Since afaik this is the main development fork for now.
Though, I don‘t think we should merge VU into PRoCon. VU isn’t something official and is just differently from the original version. Also I don’t know how the normal users (without VU) would feel about it.

procon-plugin committed:
Update uscPlayerListPanel.cs (#9)

player list panel with colored reserved slot players

procon-plugin committed:
Update CountryLookup.cs (#10)

based on settings use geoip.dat file to check player country

procon-plugin committed:
Update GlobalSuppressions.cs (#11)

new setting to enable/disable geoip.dat file and rss feed news

procon-plugin committed:
Update OptionsSettings.cs (#12)

setting for geoip.dat. setting for rss feed. updated player links for player panel

procon-plugin committed:
Update PRoConApplication.cs (#13)

use geoip file based on settings. show rss feed news based on settings. send usage data based on settings.

Prophet731 committed:
Code error fix

Prophet731 committed:
v1.5.3.5

Mik4sa commented:
Closing due to no reaction.
